Add "no statsheet" option

cwisniew opened the pull request:
### Identify the Bug or Feature request
#5643

### Description of the Change
Added a No StatSheet option and cleaned up the stat-sheet location combo box on the campaign properties and edit toke dialogs so that it is disabled when the location can't be set.

### Possible Drawbacks

Should be none

### Documentation Notes
Its now possible to set the stat sheet type as "No Stat Sheet"

### Release Notes
Its now possible to set the stat sheet type as "No Stat Sheet"

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/RPTools/maptool/5788)
<!-- Reviewable:end -->

cwisniew committed:
Add "no statsheet" option

kwvanderlinde commented:
The *Stat Sheet Location* field in the **New Token** dialog should also be disabled when set to "No Stat Sheet":
<img width="612" height="360" alt="image" src="https://github.com/user-attachments/assets/e8506c78-2049-48c8-a0ce-4b56ec766c43" />

I also think it would make sense if the "No Stat Sheet" option would be presented first in the various combo boxes rather than being sorted alphabetically with the rest.

cwisniew committed:
Fix new token dialog and combobox ordering

cwisniew commented:
> The _Stat Sheet Location_ field in the **New Token** dialog should also be disabled when set to "No Stat Sheet": <img alt="image" width="612" height="360" src="https://private-user-images.githubusercontent.com/7492219/490719455-e8506c78-2049-48c8-a0ce-4b56ec766c43.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgyMDExOTEsIm5iZiI6MTc1ODIwMDg5MSwicGF0aCI6Ii83NDkyMjE5LzQ5MDcxOTQ1NS1lODUwNmM3OC0yMDQ5LTQ4YzgtYTBjZS00YjU2ZWM3NjZjNDMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MThUMTMwODExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmUxMzRmYWI4ZDE4MjdiZWQ1MDU4NDJhYTI0MWFkMmM1NzMyZGYxMzI5NGQzZmZiOTRkNTM2NTczZjY4MWRhYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.v1BZblH_sddgBYc1taLQWBoCf2jhK-Y6DvH7iKfNn_E">
> 
> I also think it would make sense if the "No Stat Sheet" option would be presented first in the various combo boxes rather than being sorted alphabetically with the rest.

Doh, had the preference to display that dialog off so I missed it, brain is getting really rusty, thanks for picking it up.

I have updated that dialog and changed all 3 places (new token dialog, campaign properties, and edit token dialog) so that the stat sheets are ordered in such a way that all internal stat sheets (default, legacy, no) appear in the list before non internal stat sheets.
